routers/web/events/events.go

```diff
@@ -99,6 +100,14 @@ loop:
 				log.Error("Unable to GetUserStopwatches: %v", err)
 				continue
 			}
+			if len(sws) == 0 {
+				if prevStopwatchEmpty {
+					continue
+				}
+				prevStopwatchEmpty = true
+			} else {
+				prevStopwatchEmpty = false
+			}
```

Comment:
```suggestion
			if prevStopwatchEmpty {
				continue
			}
			prevStopwatchEmpty = len(sws) == 0
```

Reply:
This would be incorrect, once `prevStopwatchEmpty` is true, it can never become `false` again.

Reply:
> This would be incorrect, once `prevStopwatchEmpty` is true, it can never become `false` again, which should be in case a new stopwatch would popup.

Reply:
But... doesn't your approach have the same problem?

Reply:
My approach first checks the len of sws before it proceeds to set any value to `prevStopwatchEmpty`, it's able to go to `true` -> `false` if the users has started a new stopwatch. With the suggested approach it will never be able to do that as the check is after `prevStopwatchEmpty`

Reply:
TBH if you're gonna spend time improving the performance of this you should do a polling stopwatch loop similar to the way in which the notification loop works. This code was simply meant as a staging post before we did that fixing some far more inefficient code that involved repeated request polling instead.

Reply:
Where can I find this "notifcation loop"?

Reply:
modules/eventsource/manager_run.go

Reply:
Thank you, loop is now even better.